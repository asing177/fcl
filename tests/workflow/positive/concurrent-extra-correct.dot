digraph workflow {
edge [color="#000000"]
graph [bgcolor=transparent]


initial [shape=point; width=0.3; style=filled; color="#0e64ce"; label=""]
doneAlice [shape=ellipse; fontname="Arial"; fontsize=16; style=filled; color=black; fillcolor=white;]
doneBob [shape=ellipse; fontname="Arial"; fontsize=16; style=filled; color=black; fillcolor=white;]
final [shape=ellipse; fontname="Arial"; fontsize=16; style=filled; color=black; fillcolor=white;]
todoAlice1 [shape=ellipse; fontname="Arial"; fontsize=16; style=filled; color=black; fillcolor=white;]
todoAlice2 [shape=ellipse; fontname="Arial"; fontsize=16; style=filled; color=black; fillcolor=white;]
todoBob1 [shape=ellipse; fontname="Arial"; fontsize=16; style=filled; color=black; fillcolor=white;]
todoBob2 [shape=ellipse; fontname="Arial"; fontsize=16; style=filled; color=black; fillcolor=white;]
terminal [shape=point; width=0.3; peripheries=2; style=filled; color="#d11010"; label=""]

1[label=<<FONT POINT-SIZE="16">requestValues</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]
2[label=<<FONT POINT-SIZE="16">alternativeAlice</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]
3[label=<<FONT POINT-SIZE="16">shorterCut</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]
4[label=<<FONT POINT-SIZE="16">shortCut</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]
5[label=<<FONT POINT-SIZE="16">setValueAlice</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]
6[label=<<FONT POINT-SIZE="16">setValueBob1</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]
7[label=<<FONT POINT-SIZE="16">setValueBob2</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]
8[label=<<FONT POINT-SIZE="16">calculateTotal</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]
9[label=<<FONT POINT-SIZE="16">finish</FONT><FONT POINT-SIZE="10" COLOR="blue"> </FONT>>shape=box; fontname="Arial"; style=filled; color=black; fillcolor=gray75;]

initial -> 1 ; 1 -> {todoAlice1, todoBob1}
initial -> 2 ; 2 -> todoAlice2
todoAlice1 -> 3 ; 3 -> doneAlice
todoAlice2 -> 4 ; 4 -> final
todoAlice1 -> 5 ; 5 -> doneAlice
todoBob1 -> 6 ; 6 -> todoBob2
todoBob2 -> 7 ; 7 -> doneBob
{doneAlice, doneBob} -> 8 ; 8 -> final
final -> 9 ; 9 -> terminal

{rank=same doneAlice doneBob}
{rank=same todoAlice1 todoBob1}


}
