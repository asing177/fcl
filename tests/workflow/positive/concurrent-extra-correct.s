@initial
requestValues() {
  transitionTo(@{todoAlice1, todoBob1});
}

@initial
alternativeAlice() {
  transitionTo(@todoAlice2)
}

@todoAlice1
shorterCut() {
  transitionTo(@doneAlice)
}

@todoAlice2
shortCut() {
  transitionTo(@final)
}

@todoAlice1
setValueAlice(int val) {
  transitionTo(@doneAlice);
}

@todoBob1
setValueBob1(int val) {
  transitionTo(@todoBob2);
}

@todoBob2
setValueBob2(int val) {
  transitionTo(@doneBob);
}

@{doneAlice, doneBob}
calculateTotal() {
  transitionTo(@final);
}

@final
finish() {
  terminate()
}
