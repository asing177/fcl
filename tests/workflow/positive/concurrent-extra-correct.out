 
Method arguments and effects:
requestValues: () -> {}
alternativeAlice: () -> {}
shorterCut: () -> {}
shortCut: () -> {}
setValueAlice: (int) -> {}
setValueBob1: (int) -> {}
setValueBob2: (int) -> {}
calculateTotal: () -> {}
finish: () -> {write via terminate}